llvm/lib/Transforms/Vectorize/SLPVectorizer.cpp

```diff
@@ -19123,6 +19123,14 @@ bool SLPVectorizerPass::vectorizeStoreChains(BoUpSLP &R) {
     if (V->getPointerOperandType()->getTypeID() >
         V2->getPointerOperandType()->getTypeID())
       return false;
+    if (V->getValueOperand()->getType()->getScalarSizeInBits() <
```

Comment:
I think you also need to update the checks in AreCompatibleStores lambda

Reply:
done

Reply:
I would suggest to compare getTypeID() rather than size here

Reply:
getTypeID has been compared above, but with the same TypeID, such as IntegerTyID, which can have different sizes.

Reply:
No, getTypeID() above compared for pointer operands, which worked for typed pointers. You need to use getTypeID for value operand types

Reply:
Did you miss lines 19114 to 19122, which compare the value operand types?

Reply:
Ah, yes. Then there is a question, of why it did not work, because it should work correctly for compatible types

Reply:
i8 and i64 are not compatible types, but they have the same Type ID, IntegerTyID

Reply:
Ah, yes, missed it.